addon/validators/number.js

```diff
@@ -49,25 +49,20 @@ function _validateType(type, opts, numValue, key) {
 
 export default function validateNumber(options = {}) {
   return (key, value) => {
-    let numValue = Number(value);
-    let optKeys = keys(options);
-
-    if (options.allowBlank && isEmpty(value)) {
+    if (isEmpty(value) && options.allowBlank) {
       return true;
     }
 
-    if (typeof(value) === 'string' && isEmpty(value)) {
-      return buildMessage(key, 'notANumber', value, options);
-    }
+    let numValue = Number(value);
 
-    if(!_isNumber(numValue)) {
+    if (isEmpty(value) || !_isNumber(numValue)) {
       return buildMessage(key, 'notANumber', value, options);
-    }
-
-    if(isPresent(options.integer) && !_isInteger(numValue)) {
+    } else if (isPresent(options.integer) && !_isInteger(numValue)) {
```

Comment:
Decided against a `typeof value !== 'number'` check, since `NaN` is a number but `"8"` wouldn't be even though it should be allowed (meaning those each add an additional check)

Hopefully the intent of each set of checks now is more explicit than the last commit